test/e2e/mhc_remediations_test.go

```diff
@@ -0,0 +1,39 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("When testing unhealthy machines remediation", func() {
```

Comment:
I see that other tests have been doing this "when testing" `Describe` but this seems weird. You usually name tests for the feature, like maybe 
```suggestion
var _ = Describe("Unhealthy machine remediation", func() {
```

Reply:
@fabriziopandini WDYT?

Reply:
I'm ok with either approach, but we should stay consistent across all test.
NB. Those changes will have some impacts on test grid, but I consider this acceptable now

Reply:
It looks like this in prow:
```
When testing Cluster API working on self-hosted clusters 
  Should pivot the bootstrap cluster to a self-hosted cluster
  /home/prow/go/src/sigs.k8s.io/cluster-api/test/e2e/self_hosted.go:75
```

This looks okay to me. We can address it in another issue if you think naming needs to be changed @benmoss